forte/data/converter/converter.py

```diff
@@ -0,0 +1,98 @@
+#  Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Tuple, Any, Optional
+
+import torch
+from torch import Tensor
+
+from forte.data.converter.feature import Feature
+
+
+class Converter:
+    """
+    This class has the functionality of converting a batch of Feature to a
+    PyTorch tensor. It will also do the padding for the given batch of data.
+    """
+
+    def __init__(self, default_dtype: torch.dtype = torch.long):
+        self._default_dtype = default_dtype
+
+    def convert(self, features: List[Feature]) -> \
+            Tuple[Tensor, List[Tensor]]:
+        """
+        Convert a list of Features to a Tensor. Internally it will use
+        breadth-first search to pad all features and get the actual data and
+        corresponding masks.
+        Args:
+            features (List):
+                A list of features, where each feature can be the value or
+                another list of features.
+        Returns: A tuple containing two elements. The first element is a Tensor
+        representing the padded batch of data. The shape will be
+        [batch_size, feature_dim1_max, feature_dim2_max, ...].
+        The second element is a list of Tensors `masks` representing masks
+        along different feature dimensions. For example, the masks[i] is the
+        mask tensor along ith dimension and will have the shape:
+        [batch_size, feature_dim1_max, ..., feature_dimi_max]
```

Comment:
Maybe explain what is the mask doing. 0 for padded part, right?

Reply:
Done